### ISSUE TITLE
Tiles mapped on world before upgrade to 1.13/.1 will not render or update in dynmap 3.0. 

### ISSUE THREAD

**Issue by Electrograv** [webbukkit/dynmap#2313]
I am having an issue with the latest update to Craftbukkit 1.13 and dynmap on my 5 year old server where the plugin seems to only recognize and render areas I have been to since the update to 1.13. When I revisit the areas _after_ the update that have been explored _before_ the update, they are unchanged in-game, so clearly all of the old terrain data still exists in my world folder. I have tried commands such as **/dynmap updaterender**, and **/dynmap fullrender** and only updates regions explored. The latter command has erased the old terrain tiles in the 'flat' map I tested it on and left all but the recent areas I explored. I am concerned that I would have to revisit all previously explored terrain in order to update the maps to the new colours and style. (not to mention the 'flat' map I lost) As of now, my 'surface' map remains intact but shows ugly swaths of manually updated terrain cutting through old terrain leaving black pixels at the edges.

Is this a permanent issue? or is there some way to fix this?

P.S. Thanks for creating this great plugin! :)

<img width="1475" alt="screen shot 2018-08-31 at 2 01 26 pm" src="https://user-images.githubusercontent.com/42873427/44928668-8239f900-ad26-11e8-9e32-631ce16ecd17.png">
<img width="1400" alt="screen shot 2018-08-31 at 2 01 58 pm" src="https://user-images.githubusercontent.com/42873427/44928672-849c5300-ad26-11e8-992d-9d4197029067.png">

**Comment by smmmadden** [comment]
I believe this is expected behavior since no 1.13/1.13.1 server version is "production-ready" to support upgrading previously created worlds (e.g. 1.12.2 or lower).  Right now, all the effort has been testing on new worlds (due to the 1.13 update changing virtually everything).  At some point, upgrading worlds may be supported - though personally, I won't be moving any of the old maps to the 1.13 release.  I'll start with newly generated 1.13.1 worlds once it is ready. :-)  Many other server owners are planning the same (at least for now).  There is a lengthy discussion here about this and other issues: https://github.com/webbukkit/dynmap/issues/2271

**Comment by Psy-Virus** [comment]
You can delete the tiles of the whole world with `dynmap purgeworld <worldname>` Then you can simply do a fullrender and everything old should be gone.

**Comment by kiskoza** [comment]
@Psy-Virus not exactly. I experienced the same problem. It's not about "not re-redenring those tiles", it's "not rendering those tiles at all".

As far as I know 1.13/1.13.1 has a new chunk format and the newset versions of dynmap only renders chunks upgraded to 1.13 format and skips everything older. However the server only care to upgrade when you visit them. So you either lost your less visited areas from the map or keep the not matching old renders

I can see two possible solutions:
a) get a tool that automatically upgrades every existing chunks (maybe future versions of the server?)
b) support old chunk format in dynmap and render it the same way as the new ones

**Comment by ssshake** [comment]
I'm also having this issue and although pointed out as like, "not really a problem and deal with it" (my interpretation). It would be nice to have a fix for this. I have a very large explored area of a world which we intend to persist. We upgraded to 1.13 under the compromise that our old chunks will not have the under the sea biome but new ones would. However we didn't expect that this would cause our map to break. Very unfortunate.

**Comment by ssshake** [comment]
> Even after upgrading a whole world using the --forceupgrade parameter when spigot starts doesnt seem to fix the issue, so not sure if its related to upgrading old chunks

You tried a full render after the upgrade?

**Comment by menelmir** [comment]
So, as this is issue is not resolved, I made a very little plugin witch refresh all chunks in all worlds.
And... IT'S WORKS ! Waiting for this issue to be resolved, it's a a temporary solution.
It's provided as is and comes with no warranty, save your server first.
Make sure your read the joigned readme.txt file !!!
Just a command to use, from console.

[ChunksRefresher_0_0_1-SNAPSHOT-2018-11-02.zip](https://github.com/webbukkit/dynmap/files/2568504/ChunksRefresher_0_0_1-SNAPSHOT-2018-11-02.zip)

**Comment by menelmir** [comment]
So, as this is issue is not resolved, I made a very little plugin witch refresh all chunks in all worlds.
And... IT'S WORKS ! Waiting for this issue to be resolved, it's a a temporary solution.
It's provided as is and comes with no warranty, save your server first.
Make sure your read the joigned readme.txt file !!!
Just a command to use, from console.

NB : there can exist issues on some case in large worlds... Beware...

[ChunksRefresher_0_0_1-SNAPSHOT-2018-11-02.zip](https://github.com/webbukkit/dynmap/files/2568499/ChunksRefresher_0_0_1-SNAPSHOT-2018-11-02.zip)

**Comment by mikeprimm** [comment]
I've identified a Spigot bug that is causing this issue (basically, the loadChunk() API fails if used to load a pre-1.13 chunk without specifying the 'generate-if-needed' flag, which is a problem for dynmap since it would result in unbounded growth of the world map....).  I've coded a workaround in the recent dev builds - please check https://dynmap.us/builds/dynmap/Dynmap-3.0-SNAPSHOT-spigot.jar and let us know if it works for you

**Comment by ssshake** [comment]
I just did a fullrender of surface and this didn't fix the problem. Maybe prior work I did put me an unrecoverable state?

I'm doing a fullrender of flat now to see what happens

**Comment by mikeprimm** [comment]
The workaround is 1.13.2 specific - just be sure you are running that.   If you really need a 1.13 or 1.13.1, I can try to look in to it.

**Comment by ssshake** [comment]
💯 this appears to be working as I watch it rebuild. Thanks!
